### PR TITLE
prefect-dbt - update integration docs

### PR DESCRIPTION
**Pull request by seanpwlms**
Documents new features introduced in 0.5.0 and reorganizes content for clarity.

### Checklist

- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
- [ ] This pull request references any related issue by including "closes `<link to issue>`"
  
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: docs/integrations/prefect-dbt/index.mdx [hunk 1]
```diff
@@ -2,7 +2,7 @@
 title: prefect-dbt
 ---
 
-With prefect-dbt you can trigger and observe dbt Cloud jobs, execute dbt Core CLI commands, and incorporate other tools, such as Snowflake, into your dbt runs.
+With `prefect-dbt` you can trigger and observe dbt Cloud jobs, execute dbt Core CLI commands, and incorporate other tools, such as Snowflake, into your dbt runs.
```
Comment (billpalombi) [inline docs/integrations/prefect-dbt/index.mdx:5]:
```suggestion
With `prefect-dbt`, you can trigger and observe dbt Cloud jobs, execute dbt Core CLI commands, and incorporate other tools, such as [Snowflake](integrations/prefect-snowflake/index), into your dbt runs.
```

File: docs/integrations/prefect-dbt/index.mdx [hunk 2]
```diff
@@ -68,7 +67,58 @@ def run_dbt_job_flow():
 run_dbt_job_flow()
 ```
 
-## Integrate dbt Core CLI commands with Prefect flows
+### Save dbt Cloud credentials to a block
+
+Blocks can be [created through code](/3.0rc/develop/connect-third-party/#saving-blocks) or through the UI.
+
+
+To create a dbt Cloud Credentials block do the following:
```
Comment (billpalombi) [inline docs/integrations/prefect-dbt/index.mdx:75]:
```suggestion
To create a dbt Cloud Credentials block:
```

File: docs/integrations/prefect-dbt/index.mdx [hunk 3]
```diff
@@ -68,7 +67,58 @@ def run_dbt_job_flow():
...
+
+1. Go to your [dbt Cloud profile](https://cloud.getdbt.com/settings/profile).
+2. Log in to your dbt Cloud account.
+3. Scroll to **API** or click **API Access** on the sidebar.
+4. Copy the API Key.
+5. Click **Projects** on the sidebar.
+6. Copy the account ID from the URL: `https://cloud.getdbt.com/settings/accounts/<ACCOUNT_ID>`.
+7. Create and run the following script, replacing the placeholders.
```
Comment (billpalombi) [inline docs/integrations/prefect-dbt/index.mdx:83]:
```suggestion
7. Create and run the following script, replacing the placeholders:
```

File: docs/integrations/prefect-dbt/index.mdx [hunk 4]
```diff
@@ -68,7 +67,58 @@ def run_dbt_job_flow():
...
+```python
+from prefect_dbt.cloud import DbtCloudCredentials
+
+DbtCloudCredentials(
+    api_key="API-KEY-PLACEHOLDER",
+    account_id="ACCOUNT-ID-PLACEHOLDER"
+).save("CREDENTIALS-BLOCK-NAME-PLACEHOLDER")
+```
+
+Then, to create a dbt Cloud job block do the following:
```
Comment (billpalombi) [inline docs/integrations/prefect-dbt/index.mdx:94]:
```suggestion
Then, create a dbt Cloud job block:
```

File: docs/integrations/prefect-dbt/index.mdx [hunk 5]
```diff
@@ -100,11 +150,75 @@ if __name__ == "__main__":
     trigger_dbt_flow()
 ```
 
-### Set up a new profile
 
-To setup a new profile, first [save and load a DbtCliProfile block](#saving-credentials-to-block) and use it in `DbtCoreOperation`.
+If you are already using Prefect blocks such as the Snowflake Connector block, you can use those blocks to [create a new profiles.yml with a DbtCliProfile block](#create-a-new-profile-with-blocks).
```
Comment (billpalombi) [inline docs/integrations/prefect-dbt/index.mdx:154]:
```suggestion
If you are already using Prefect blocks such as the [Snowflake Connector block](integrations/prefect-snowflake/index#access-underlying-snowflake-connection), you can use those blocks to [create a new profiles.yml with a DbtCliProfile block](#create-a-new-profile-with-blocks). 
```

File: docs/integrations/prefect-dbt/index.mdx [hunk 6]
```diff
@@ -100,11 +150,75 @@ if __name__ == "__main__":
...
 
-### Set up a new profile
 
-To setup a new profile, first [save and load a DbtCliProfile block](#saving-credentials-to-block) and use it in `DbtCoreOperation`.
+If you are already using Prefect blocks such as the Snowflake Connector block, you can use those blocks to [create a new profiles.yml with a DbtCliProfile block](#create-a-new-profile-with-blocks). 
+
+
+#### Use environment variables with Prefect secret blocks
+
+If you use environment variables in `profiles.yml`, set a Prefect Secret block as an environment variable.
```
Comment (billpalombi) [inline docs/integrations/prefect-dbt/index.mdx:159]:
```suggestion
If you use environment variables in `profiles.yml`, set a Prefect Secret block as an environment variable:
```

**Comment by seanpwlms** [comment]
Thanks for the feedback, @billpalombi

File: docs/integrations/prefect-dbt/index.mdx [hunk 7]
```diff
@@ -100,11 +150,75 @@ if __name__ == "__main__":
...
+        project_dir="/Users/test/my_dbt_project_dir",
+        create_summary_artifact: bool = True,
+        summary_artifact_key: str = "dbt-build-task-summary",
+        extra_command_args=["--model", "foo_model"]
+    )
+if __name__ == "__main__":
+    dbt_build_flow()
+```
+
+To see all of the pre-built tasks, visit the SDK docs.
```
Comment (billpalombi) [inline docs/integrations/prefect-dbt/index.mdx:209]:
```suggestion
See the [SDK docs](https://prefect-python-sdk-docs.netlify.app/prefect_dbt/) for other pre-built tasks.
```

**Pull request merged**